crates/evm/src/execute.rs

```diff
@@ -290,7 +293,7 @@ impl<F> BasicBlockExecutorProvider<F> {
 
 impl<F> BlockExecutorProvider for BasicBlockExecutorProvider<F>
 where
-    F: BlockExecutionStrategyFactory,
+    F: BlockExecutionStrategyFactory<Error = BlockExecutionError>,
```

Comment:
```suggestion
    F: BlockExecutionStrategyFactory,
```
the error type can't be hardcoded here, it needs to work for both `OpBlockExecutionError` and `ExecutionError`